pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -343,8 +336,14 @@ func (m *rmMigration) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
 		return err
 	}
 
-	if err := os.RemoveAll(silencesFileName(mg)); err != nil {
-		mg.Logger.Error("alert migration error: failed to remove silence file", "err", err)
+	files, err := getAllSilenceFiles(mg)
```

Comment:
Remind me again why do we remove the silences file?

Reply:
This task runs during [the backward migration](https://github.com/grafana/grafana/blob/90572dc7219ae3fdaeaa46f6a133c1e9b60479b5/pkg/services/sqlstore/migrations/ualert/ualert.go#L68) (when disabling `ngalert`) so we clean everything related to the new alerting.